activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -36,9 +36,15 @@ module Calculations
     # Note: not all valid +select+ expressions are valid +count+ expressions. The specifics differ
     # between databases. In invalid cases, an error from the database is thrown.
     def count(column_name = nil, options = {})
+      if options.present? && !ActiveRecord.const_defined?(:DeprecatedFinders)
+        raise ArgumentError, "Relation#count does not support finder options anymore. " \
+                             "Please build a scope and then call count on it or use the " \
+                             "activerecord-deprecated_finders gem to enable this functionality."
```

Comment:
Hey @kaspth, this is targeted to 4.2 only, not master! Its purpose being to ease upgrades from 4.0 to more recent versions of 4.x. See also https://github.com/rails/rails/issues/20434 for background/context.